api/test/wfl/integration/modules/covid_test.clj

```diff
@@ -100,3 +118,42 @@
                   (workloads/covid-workload-request {} {} {}))]
     (is (not (:started workload)))
     (is (:started (workloads/start-workload! workload)))))
+
+(deftest test-update-terra-workspace-sink
+  (fixtures/with-temporary-workspace
+    workspace-prefix group
+    (fn [workspace]
+      (let [flowcell-id
+            "test"
+            workflow
+            {:uuid    "2768b29e-c808-4bd6-a46b-6c94fd2a67aa"
+             :status  "Succeeded"
+             :outputs (-> "sarscov2_illumina_full/outputs.edn"
+                          resources/read-resource
+                          (assoc :flowcell_id flowcell-id))}
+            executor
+            (make-queue-from-list [workflow])
+            sink
+            (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+              (->> {:name           "Terra Workspace"
+                    :workspace      workspace
+                    :entity         "flowcell"
+                    :fromOutputs    (resources/read-resource
+                                     "sarscov2_illumina_full/entity-from-outputs.edn")
+                    :identifier     "flowcell_id"
+                    :skipValidation true}
+                   (covid/create-sink! tx 0)
+                   (zipmap [:sink_type :sink_items])
+                   (covid/load-sink! tx)))]
+        (covid/update-sink! executor sink)
+        (is (-> executor :queue empty?) "The workflow was not consumed")
+        (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+          (let [records (->> sink :details (postgres/get-table tx))]
+            (is (== 1 (count records))
+                "The record was not written to the database")
+            (is (= (:uuid workflow) (-> records first :workflow))
+                "The workflow UUID was not written")))
+        (let [[{:keys [name]} & _]
+              (util/poll
+               (fn [] (seq (firecloud/list-entities workspace "flowcell"))))]
+          (is (= name flowcell-id) "The test entity was not created"))))))
```

Comment:
Where does `"sarscov2_illumina_full/outputs.edn"` (and so on) come from?
Who or what generates it?
`"sarscov2_illumina_full/entity-from-outputs.edn"` looks like `clojure.core/name`.

Reply:
So entity-from-outputs.edn is https://github.com/broadinstitute/horsefish/blob/master/scripts/cdc-sabeti-covid-19/batch_upsert_entities.py encoded as data?

Reply:
Pretty much yeah